lib/draft/feature_test_phoenix_integration.md

```diff
@@ -0,0 +1,232 @@
+
+> as a teacher I would like to be able to create contracts for my students so that I don't have to worry about getting paid
+
+## So why is this distinction important?
+
+The main reason I think this distinction is important is this. When we know the true intent of what we are trying to solve, we might find other ways to solve that particular problem. To use this example, if the problem a teacher has is getting paid because they are constantly chasing their money, we could also solve that problem by creating a subscription based system. So the business owner asked for contracts because that is what teachers normally do when teaching, however that might not be the only option. For now I am going with this as I think it is the more simple approach to solve this problem. Building a subscription system is something I will probably tackle in the future, but I'd like to make sure we provide business value in small iterations and the first iteration for now will be about creating these contracts.
+
+## Let's write some Gherkin
+
+With one user story I often find myself writing a whole range of acceptance criteria using Gherkin. Let's start here with defining a few to get us going.
+
+```gherkin
+# Contract Listing
+GIVEN I have a teacher account at studentmanager
+AND I log in
+AND I have an existing student
+AND that student has a contract
+WHEN I visit the page for this student
+THEN I should see a list of contracts for this student
+AND should see the specific contract for this student
+
+# Adding a contract
+GIVEN I am looking at the details for a student
+WHEN I click on Add Contract
+THEN I should be able to enter all the details needed to create a contract 
+
+# Contract created flow
+GIVEN I Have entered all the details for a contract for a student
+WHEN I create this contract
+THEN I should be redirected back to the students overview page
+AND I should see the contract just created
+AND the students parent should receive an email with the details of this new contract
+
+# Contract approval flow
+GIVEN I am a parent of a student
+AND I have received an email indicating there is a new contract
+WHEN I click the link provided in the email
+AND I log in after following that link
+THEN I should see a page where I can view the contract details
+AND I should be able to accept and approve the contract 
+```
+
+So that's a good start. If you notice (like above here) that you have a lot of acceptance criteria for one story, it is usually a sign that it's probably best to split it up into more user stories. This way the original user story can become an Epic and you can add all the more fine-grained stories as part of this Epic. 
```

Comment:
> a sign that it's probably best to split it up into more user stories. 

Do you want to show this the reader as well, because I think that this is important - Maybe high-level listing of the epic - stories division.